### PR TITLE
Display error message for occupied positions

### PR DESCRIPTION
**Pull request by paul90317**
Display an error message when a player attempts to place a marker (circle/cross) on a position already occupied by another marker.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Read https://cbea.ms/git-commit/ carefully and refine the git commit message.

File: ttt.c [hunk 1]
```diff
@@ -443,9 +443,16 @@ int main()
         } else {
             draw_board(table);
             int move;
-            do {
+            while (1) {
                 move = get_input(turn);
-            } while (table[move] != ' ');
+                if (table[move] == ' ') {
+                    break;
+                } else {
```
Comment (jserv) [inline ttt.c]: The `else` statement here is not needed since the previous one is `break`.

**Comment by paul90317** [comment]
It seems that I forgot to wrap the body at 72 characters. I have remade the commit message.

File: ttt.c [hunk 2]
```diff
@@ -443,9 +443,15 @@ int main()
...
             int move;
-            do {
+            while (1) {
                 move = get_input(turn);
-            } while (table[move] != ' ');
+                if (table[move] == ' ') {
+                    break;
+                }
+                printf(
+                    "Invalid operation: O/X already placed at this "
```
Comment (jserv) [inline ttt.c]: Improve this message. Check https://en.wikipedia.org/wiki/Tic-tac-toe for proper rule description.

**Review by jserv — CHANGES_REQUESTED** [review]
Refine the Git commit message by formulating it as an unambiguous and complete English sentence.

**Comment by jserv** [comment]
Thank @paul90317 for contributing!

**Pull request merged**